apollo-integration/src/test/java/com/apollographql/android/impl/ApolloWatcherTest.java

```diff
@@ -234,6 +234,58 @@ public void rewatchCacheControl() throws IOException, InterruptedException {
     secondResponseLatch.await();
   }
 
+  @Test
+  public void testQueryWatcherUpdated_SameQuery_DifferentResults_cacheOnly() throws IOException, InterruptedException {
+    final CountDownLatch cacheWarmUpLatch = new CountDownLatch(1);
+    EpisodeHeroName query = new EpisodeHeroName(EpisodeHeroName.Variables.builder().episode(Episode.EMPIRE).build());
+    server.enqueue(mockResponse("EpisodeHeroNameResponseWithId.json"));
+    apolloClient.newCall(query).enqueue(new ApolloCall.Callback<EpisodeHeroName.Data>() {
+      @Override public void onResponse(@Nonnull Response<EpisodeHeroName.Data> response) {
+        cacheWarmUpLatch.countDown();
+      }
+
+      @Override public void onFailure(@Nonnull Exception e) {
+        Assert.fail(e.getMessage());
+        cacheWarmUpLatch.countDown();
+      }
+    });
+    cacheWarmUpLatch.await();
+
+    //Cache is now "warm" with response data
+
+    final CountDownLatch firstResponseLatch = new CountDownLatch(1);
+    final CountDownLatch secondResponseLatch = new CountDownLatch(2);
+
+    ApolloWatcher<EpisodeHeroName.Data> watcher = apolloClient.newCall(query)
```

Comment:
Is only way to set cache policy per request? Is it possible to set default cache policy for all requests?